include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -400,7 +381,19 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         }
 
     private:
-        KOKKOS_INLINE_FUNCTION int find_cell(ddc::Coordinate<Tag> const& x) const;
+        KOKKOS_INLINE_FUNCTION discrete_element_type
+        get_first_bspline_in_cell(ddc::DiscreteElement<knot_mesh_type> const& ic) const
+        {
+            return discrete_element_type((ic - m_break_point_domain.front()).value());
+        }
+
+        /**
+         * @brief Get the cell where x is found
+         * @param x The point whose location must be determined.
+         * @returns The DiscreteElement describing the knot at the lower bound of the cell of interest.
+         */
+        KOKKOS_INLINE_FUNCTION ddc::DiscreteElement<knot_mesh_type> find_cell(
+                ddc::Coordinate<Tag> const& x) const;
```

Comment:
Should we rename it ? like `get_closest_knot_before` ? Because we don't return the cell element but a knot element ?

Reply:
Maybe `find_cell_start`?

Reply:
7a559806